sygnal/apnspushkin.py

```diff
@@ -273,6 +294,7 @@ async def _dispatch_notification_unlimited(self, n, device, context):
                     with self.sygnal.tracer.start_span(
                         "apns_dispatch_try", tags=span_tags, child_of=span_parent
                     ) as span:
+                        assert shaved_payload is not None
```

Comment:
This feels like something that we should avoid being necessary

Reply:
Actually yes: why *is* this necessary :o

Reply:
So the reasoning behind this was: `shaved_payload` is assigned to the return value of `apnstruncate.truncate` a few lines above this assertion, and the return value of `apnstruncate.truncate` is currently `Optional[Dict[str, Any]`, so mypy got upset that we were potentially passing a `None` into `_dispatch_request`. The mypy docs said to add an `assert foo is not none` in those types of cases, so that is what I did, although it seems like I should have handled it in another way.

Reply:
And this is all because I was confused about the return type of `apnstruncate.truncate` which is wrong.